pkg/clusterfile/clusterfile.go

```diff
@@ -146,6 +153,18 @@ func (c *ClusterFile) SaveAll() error {
 	return utilsos.NewCommonWriter(fileName).WriteFile(bytes.Join(clusterfileBytes, []byte("---\n")))
 }
 
+func (c *ClusterFile) RollBackClusterFile(err error) error {
+	//if there exits an error,rollback the ClusterFile to the default file
+	if err == nil {
+		return nil
+	}
+	c.cluster = c.cfSnapshot
```

Comment:
This makes "cluster" and "cfSnapshot" refer to a same address.
What if we don't exit process after RollBackClusterFile()?
After "RollBackClusterFile", what we do to "cluster" will also impact "cfSnapshot".
We'd better clone "cfSnapshot", and give it to "cluster".
And check "cfSnapshot == nil"

Reply:
Thanks,I'll optimsize them.